### ISSUE TITLE
Allow Scala to have mixed Symbol/Constant

### ISSUE THREAD

**Issue by lanking520** [apache/incubator-mxnet#13403]
[Quote from Email]
I'm trying to figure out how to do something like this
https://github.com/ufoym/mxnet/blob/master/example/vae/VAE.py#L83, however
it seems the Scala Symbol API does not allow the mixing of symbols and constants like the python library does.

 It seems like if I want to use constants in my loss functions I'm going to have to have a very large argsDict when I go to train since every variable in the loss function definition will have to be symbolic. Is there a better way to do this?

-- Sam

**Comment by lanking520** [comment]
@mxnet-label-bot please add [Scala, Question]

**Comment by lanking520** [comment]
Hi Sam, about the line you have pointed:
```
z = mu + mx.symbol.broadcast_mul(mx.symbol.exp(0.5*logvar),
                                         mx.symbol.random_normal(loc=0, scale=1, shape=(batch_size, n_latent)))
```
I didn't see why you cannot do this. In fact, this piece of python code did not mix with constant, they are all Symbols (mu, logvar...). Could you please point to the code you have so we can help you out?

**Comment by sam-h-bean** [comment]
So in the line I pointed to there is that ```0.5``` that needs to be a symbol variable or you need to ```import org.apache.mxnet.SymbolConversions._``` to get something like this to work.

**Comment by lanking520** [comment]
> So in the line I pointed to there is that `0.5` that needs to be a symbol variable or you need to `import org.apache.mxnet.SymbolConversions._` to get something like this to work.

Thanks for your reply! The implicit methods in there should not have much diff in performance. Please let me know if you see a performance decrease after implementing it with Scala

**Comment by sam-h-bean** [comment]
No, go ahead and close this out.

**Comment by lanking520** [comment]
Thanks
